src/index.ts

```diff
@@ -87,9 +87,26 @@ export const validatorCompiler: FastifySchemaCompiler<ZodAny> =
     }
   };
 
-export const serializerCompiler: FastifySerializerCompiler<ZodAny> =
-  ({ schema }) =>
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function hasOwnProperty<T, K extends PropertyKey>(obj: T, prop: K): obj is T & Record<K, any> {
+  return Object.prototype.hasOwnProperty.call(obj, prop);
+}
+
+function resolveSchema(maybeSchema: ZodAny | { properties: ZodAny }): Pick<ZodAny, 'safeParse'> {
+  if (hasOwnProperty(maybeSchema, 'safeParse')) {
+    return maybeSchema;
+  }
+  if (hasOwnProperty(maybeSchema, 'properties')) {
+    return maybeSchema.properties;
+  }
+  throw new Error(`Invalid schema passed: ${JSON.stringify(maybeSchema)}`);
+}
```

Comment:
Relying on `properties` being present restricts schemas to objects only. It should be possible to return any type of value (i.e. array, string, boolean, number and object).

Reply:
I'll add a test for this case, thanks

Reply:
From my understanding status code may or may not be a part of zod schema, it is entirely up to whoever is designing structure of reply payload.

I've updated test suite for this change, and defining schema as a `z.string` actually seems to work perfectly fine. What doesn't work is defining schema as an object, and then returning primitive instead - since fastify won't invoke serializer at all in such case, we are not getting to validate anything. But that's a separate problem, see https://github.com/turkerdev/fastify-type-provider-zod/pull/15

Within the context of the original problem, I believe we are solving it correctly, and it should behave not worse or better than before.

Please check new tests, maybe I missed something there :)